presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -277,6 +277,7 @@ public class FeaturesConfig
     private boolean useDefaultsForCorrelatedAggregationPushdownThroughOuterJoins = true;
     private boolean mergeDuplicateAggregationsEnabled = true;
     private boolean fieldNamesInJsonCastEnabled;
+    private boolean legacyJsonCast;
```

Comment:
I'm not sure that defaulting to false is the right thing here. In particular, while it's definitely buggy to make the values lower case, the keys it's much less clear, as quotation marks in the json string aren't the same as quotation marks in sql.  They aren't evidence that the user wants to preserve the case.